packages/react-sdk/src/store/api/documentPreviewApi.ts

```diff
@@ -0,0 +1,131 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { StateEvent } from '@matrix-widget-toolkit/api';
+import { first, isError } from 'lodash';
+import { filter } from 'rxjs';
+import {
+  DocumentPreviewEvent,
+  STATE_EVENT_DOCUMENT_PREVIEW,
+  isValidDocumentPreviewEvent,
+} from '../../model';
+import { ThunkExtraArgument } from '../store';
+import { baseApi } from './baseApi';
+
+/**
+ * Endpoints to read the document preview event.
+ *
+ * @remarks this api extends the {@link baseApi} so it should
+ *          not be registered at the store.
+ */
+export const documentPreviewApi = baseApi.injectEndpoints({
+  endpoints: (builder) => ({
+    /**
+     * Return the document preview event of a room.
+     */
+    getDocumentPreview: builder.query<
+      { event: StateEvent<DocumentPreviewEvent> | undefined },
+      void
+    >({
+      queryFn: async (_, { extra }) => {
+        const widgetApi = await (extra as ThunkExtraArgument).widgetApi;
+
+        try {
+          const events = await widgetApi.receiveStateEvents(
+            STATE_EVENT_DOCUMENT_PREVIEW,
+          );
+
+          return {
+            data: { event: first(events.filter(isValidDocumentPreviewEvent)) },
+          };
+        } catch (e) {
+          return {
+            error: {
+              name: 'LoadFailed',
+              message: `Could not load document preview: ${
+                isError(e) ? e.message : e
+              }`,
+            },
+          };
+        }
+      },
+
+      async onCacheEntryAdded(
+        _,
+        { cacheDataLoaded, cacheEntryRemoved, extra, updateCachedData },
+      ) {
+        const widgetApi = await (extra as ThunkExtraArgument).widgetApi;
+
+        // wait until first data is cached
+        await cacheDataLoaded;
+
+        const subscription = widgetApi
+          .observeStateEvents(STATE_EVENT_DOCUMENT_PREVIEW)
+          .pipe(filter(isValidDocumentPreviewEvent))
+          .subscribe((event) => {
+            updateCachedData(() => ({ event }));
+          });
+
+        // wait until subscription is cancelled
+        await cacheEntryRemoved;
+
+        subscription.unsubscribe();
+      },
+    }),
+
+    /**
+     * Create a document snapshot in the current room.
+     */
+    createDocumentPreview: builder.mutation<
+      { event: StateEvent<DocumentPreviewEvent> },
+      { documentId: string; data: string }
+    >({
+      async queryFn({ data }, { extra }) {
+        const widgetApi = await (extra as ThunkExtraArgument).widgetApi;
+
+        try {
+          const maxSize = 34000;
+          // if preview data is too big, clear the preview content.
```

Comment:
I was following [our own documentation](https://github.com/nordeck/matrix-neoboard/blob/main/docs/adrs/adr005-data-structure-for-storing-whiteboards-in-matrix-rooms.md#chunking), that explains why 34kb is a good treshold for data. Because state events aren't encrypted and the data at this point is already base64 encoded, the maxSize can be further increased - maybe 60Kb can be set instead?